dataservice/api/diagnosis/schemas.py

```diff
@@ -20,6 +22,15 @@ class DiagnosisSchema(BaseSchema):
     diagnosis_category = field_for(Diagnosis, 'diagnosis_category',
                                    validate=enum_validation_generator(
                                        DIAGNOSIS_CATEGORY_ENUM))
+    mondo_id_diagnosis = FieldValidator(attribute='mondo_id_diagnosis',
+                                        validate=validate_ontology_id_prefix)
+    icd_id_diagnosis = FieldValidator(attribute='icd_id_diagnosis',
+                                      validate=validate_ontology_id_prefix)
+    uberon_id_tumor_location = FieldValidator(
+        attribute='uberon_id_tumor_location',
+        validate=validate_ontology_id_prefix)
+    ncit_id_diagnosis = FieldValidator(attribute='ncit_id_diagnosis',
+                                       validate=validate_ontology_id_prefix)
```

Comment:
maybe `validate_ontology_id_prefix` could be a generator that takes a prefix and returns a function that validates that field using the prefix? That way lookup dict won't be needed and you won't have to pass attributes to it.

Reply:
👍 I agree thats what I would suggest as well ^